tests/rest_server_test.cc

```diff
@@ -144,3 +155,104 @@ TEST(rest_server_test, response_status_code_test)
 
     stats.shutdown();
 }
+
+TEST(rest_server_test, keepalive_server_timeout)
+{
+    int thr = 1;
+    Address addr(Ipv4::any(), Port(0));
+    StatsEndpoint stats(addr);
+    stats.init(thr);
+    stats.start();
+    Port port = stats.getPort();
+
+    httplib::Client client("localhost", port);
+    client.set_keep_alive(true);
+    auto peer = stats.getAllPeer();
+
+    // first request
+    auto res = client.Get("/read/hostname");
+    ASSERT_EQ(res->status, 200);
+    peer           = stats.getAllPeer();
+    auto peerPort1 = (*peer.begin())->address().port();
+    ASSERT_EQ(peer.size(), 1);
```

Comment:
I had left another comment three hours ago, but apparently GitHub lost it. I said something like the following:

Here you use `ASSERT_EQ()` to check for conditions, but GoogleTest also offers an `EXPECT_EQ()` macro; the difference is that `ASSERT_EQ()` aborts the current test if the assertion fails, while `EXPECT_EQ()` continues running the test and reports the error at the end of the run. This makes it possible to potentially spot different failures in one single run.

So, does the failure of one of these assertions compromises also all the other ones? If not, please consider using `EXPECT_EQ()` instead, where appropriate.